snapcraft/internal/project_loader.py

```diff
@@ -299,10 +303,34 @@ def _build_env(root, arch_triplet):
         for envvar in ['CPPFLAGS', 'CFLAGS', 'CXXFLAGS']:
             env.append(formatting_utils.format_path_variable(
                 envvar, paths, prepend='-I', separator=' '))
+
+    if confinement == 'classic':
+        if not core_linker:
+            raise EnvironmentError('classic confinement requires the '
+                                   'core_linker to be set')
+
+        core_path = os.path.join('/snap', 'core', 'current')
+        core_rpaths = common.get_library_paths(core_path, arch_triplet,
+                                               existing_only=False)
+        snap_path = os.path.join('/snap', snap_name, 'current')
+        snap_rpaths = common.get_library_paths(snap_path, arch_triplet,
+                                               existing_only=False)
+
+        rpaths = formatting_utils.combine_paths(
+            core_rpaths + snap_rpaths, prepend='', separator=':')
+        env.append('LDFLAGS="$LDFLAGS '
+                   # Building tools to continue the build becomes problematic
+                   # with nodefaultlib.
+                   '-Wl,-z,nodefaultlib '
+                   '-Wl,--enable-new-dtags '
```

Comment:
I'm curious why this is used if we also remove LD_LIBRARY_PATHs from our generated environment?